pages/checkoutOverviewPage.js

```diff
@@ -0,0 +1,30 @@
+const Page = require('./page');
+const CheckoutCompletePage = require('../pages/checkoutCompletePage.js');
+const checkoutCompletePage = new CheckoutCompletePage();
+
+class CheckoutOverviewPage extends Page {
+
+    constructor() {
+        super();
+    }
+    
+    open() {
+       super.open("checkout-step-two.html");
+    }
+    injectUser(){
+        browser.execute("window.sessionStorage.setItem('session-username', 'standard-user')");                
+    }
+    injectCartContents(){
+        browser.execute("window.sessionStorage.setItem('cart-contents', '[4,1]')");
+        browser.refresh();
+    }
+    getItemCount(){
+        return parseInt(browser.$(".shopping_cart_badge").getText());
+    } 
+    finishCheckout(){
+        browser.$(".cart_button").click();
+        return checkoutCompletePage;
```

Comment:
This is a bad practice, see Titus his talk about page objects 😉 . Don't return the PO if not needed (the only case I would use it if you construct a new PO based on an element or something)

Reply:
Interesting, why is that? I don't know if I agree that

Reply:
You're now creating a fluent API by _chaining_ page objects. This PO is simple, but you could now for example also do something like this `checkoutOverviewPage.finishCheckout().injectUser()` making is more complex and easy to break.
The advice here is to keep the actions small and not chain-able which will give a better readable stacktrace and debug possibilities.